### PR TITLE
chore: nonce test

### PR DESCRIPTION
**Pull request by joepegler**
## PR-Codex overview
This PR focuses on refactoring and enhancing the functionality of the `nexusClient` operations, particularly around nonce handling and validation modes, while improving test coverage and utility functions.

### Detailed summary
- Updated `GetNonceArgs` to use string literals for `validationMode`.
- Introduced `TIMESTAMP_ADJUSTMENT` constant in `toNexusAccount`.
- Improved nonce handling in various tests and utility functions.
- Added a new test for `makeNonceKey` to validate key generation.
- Modified `prepareUserOperation` to include nonce in several test cases.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/sdk/account/toNexusAccount.ts [hunk 1]
```diff
@@ -333,31 +333,31 @@ export const toNexusAccount = async (
...
     try {
+      const TIMESTAMP_ADJUSTMENT = 1000000n
+
+      const defaultedKey = BigInt(parameters?.key ?? 0n) / TIMESTAMP_ADJUSTMENT
+      const defaultedValidationMode = parameters?.validationMode ?? "0x00"
+
       const key: string = concat([
-        "0x000000",
-        _validationMode,
+        toHex(defaultedKey, { size: 3 }),
```
Comment (livingrockrises) [inline src/sdk/account/toNexusAccount.ts:347]: this does the same job as numTo3Bytes? like others util function and ethers test

**Comment by livingrockrises** [comment]
```
const hash = await nexusClient.sendTransaction({
      calls: [
        {
          to: recipientAddress,
          value: 1n
        }
      ],
      nonce: await nexusClient.account.getNonce({key: 123456n })
    })
```

I am passing like above.
And in the logs in getNonce function it prints
params . key  1728296984990n

 If I pass
 nonce: await nexusClient.account.getNonce({key: 0n })
 
 and still it is a timestamp
 params . key  1728297053966n
 
 so no matter what we get timestamp there. even if I don't pass key at all. these are my findings
 
 now, biggest number that could fit for 3 bytes is : 16,777,215
 
 so what we are doing is dividing timestamp by 1000000 and then converting. this trims the timestamp to higher digits so there is a chance we will end up with same nonce key no? effectively making it a 1D nonce for back to back transactions.

**Comment by livingrockrises** [comment]
> ```
> const hash = await nexusClient.sendTransaction({
>       calls: [
>         {
>           to: recipientAddress,
>           value: 1n
>         }
>       ],
>       nonce: await nexusClient.account.getNonce({key: 123456n })
>     })
> ```
> 
> I am passing like above. And in the logs in getNonce function it prints params . key 1728296984990n
> 
> If I pass nonce: await nexusClient.account.getNonce({key: 0n })
> 
> and still it is a timestamp params . key 1728297053966n
> 
> so no matter what we get timestamp there. even if I don't pass key at all. these are my findings
> 
> now, biggest number that could fit for 3 bytes is : 16,777,215
> 
> so what we are doing is dividing timestamp by 1000000 and then converting. this trims the timestamp to higher digits so there is a chance we will end up with same nonce key no? effectively making it a 1D nonce for back to back transactions.

I am using % now in my branch.

**Comment by livingrockrises** [comment]
@joepegler can I close this PR?
will move the tests to my branch. 
can keep publish script in different PR

**Review by livingrockrises — APPROVED** [review]

**Review by VGabriel45 — APPROVED** [review]

**Comment by livingrockrises** [comment]
leaving the approved and merged in favour of PR # 67

we need to review ownable validator tests requiring nonce to be passed on because of getNonce function changes, and every call to getNonce brings timestamp in picture now.
https://biconomyworkspace.slack.com/archives/D069UKK1QAY/p1728064246686389?thread_ts=1728056699.180779&cid=D069UKK1QAY

**Pull request merged**